key.core/src/main/java/de/uka/ilkd/key/settings/StrategySettings.java

```diff
@@ -30,7 +32,7 @@ public class StrategySettings extends AbstractSettings {
     /**
      * maximal number of automatic rule applications before an interaction is required
      */
-    private int maxSteps = -1;
+    private int maxSteps = 10000;
```

Comment:
Why this change? 10000 is a magic number, that seems to include a decision.

Reply:
Because `-1` is a forbidden number for `maxSteps`. It was use as a temporary value and always overwritten by `readSettings`. `10000` is the default value, which is used in the case of any invalid settings.

I have made it only more explicit.